### PR TITLE
ENH: Set positions from `.gro` file

### PR DESCRIPTION
**Pull request by mattwthompson**
### Description

Beginning #1210 

### Checklist

- [ ] Add tests
- [ ] Lint
- [ ] Update docstrings

### CODE REVIEW

**Review by j-wags — APPROVED** [review]
This is approved, but it does have the fairly pointy edge case of being vulnerable to different atom orderings. It may be more trouble than it's worth to police this (since truly scrambled atom orderings will lead to massive energies and likely crashes right after) but I think the docstring at least needs to contain this warning in a way that's hard to miss. And I'd be OK with even more extreme cautions, like a UserWarning that's always omitted when the function is used unless the user sets a flag or something (though that's quite extreme so it's not a blocking requirement for merge).

File: openff/interchange/components/interchange.py [hunk 1]
```diff
@@ -903,6 +904,24 @@ def from_gromacs(
             ),
         )
 
+    def set_positions_from_gro(
+        self,
+        gro_file: Path | str,
+    ):
+        """Set the positions of this `Interchange` from a GROMACS coordinate `.gro` file."""
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply openff/interchange/components/interchange.py:911]: Great point
